WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -379,14 +431,21 @@ private boolean mediaAlreadyQueuedOrUploading(MediaModel mediaModel) {
             AppLog.i(T.MEDIA, "Looking to add media with path " + mediaModel.getFilePath() + " and site id " +
                     mediaModel.getLocalSiteId() + ". Comparing with " + queuedMedia.getFilePath() + ", " +
                     queuedMedia.getLocalSiteId());
-            if (queuedMedia.getLocalSiteId() == mediaModel.getLocalSiteId() &&
-                    StringUtils.equals(queuedMedia.getFilePath(), mediaModel.getFilePath())) {
+            if (areTheseTheSameMedia(queuedMedia, mediaModel)) {
                 return true;
             }
         }
         return false;
     }
 
+    private boolean areTheseTheSameMedia(MediaModel media1, MediaModel media2) {
```

Comment:
I think this should have a more descriptive name, maybe `compareBySiteAndFilePath`.

Also, we can simplify this method and just `return media1.getLocalSiteId() == media2.getLocalSiteId() && StringUtils.equals(media1.getFilePath(), media2.getFilePath())`.